DiscordBot.DataAccess/IEventsSheetsService.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using DiscordBot.Models;
+
+namespace DiscordBot.DataAccess
+{
+    interface IEventsSheetsService
+    {
+        Task AddEventAsync(string name, string description, DateTime time, Dictionary<string, string> roleEmotes);
+        Task EditEventAsync(int key, string? description, string? name, DateTime? time);
+        Task RemoveEventAsync(int key);
+
+        Task<DiscordEvent> GetEvent(int eventKey);
+        Task<IEnumerable<DiscordEvent>> ListEventsAsync();
+
+        Task<string> GetRoleOfEmote(int eventKey, string emote);
+
+        Task EditSignUp(int eventKey, string userId, string role, bool roleStatus);
+        Task<Dictionary<string, Dictionary<string, bool>>> GetSignUps(int key);
```

Comment:
A note for the future, it's a good idea to focus on a small part of the functionality first. For example just adding a new event. Then the workflow with be something like:
1. Add interface method(s) for new functionality
2. Use interface method(s) on the front end
3. Implement the interface method(s)

with 2 and 3 possibly being done in parallel. Even now that you've added all the interface methods I would recommend implementing it one small functionality at a time, and agreeing between you what you should work on first. FYI @kwunkiuma 🙂

Reply:
Ah ok, that's useful to know, thank you! 🙂